resources/views/standings/upload.blade.php

```diff
@@ -1,241 +1,3533 @@
+                    points: 0
+                },
+                results: [
+                    {
+                        position: 1,
+                        constructor_id: 0,
+                        grid: 0,
+                        stops: -1,
+                        time: "",
+                        fastestlaptime: "",
+                        status: -2,
+                        driver_id: 0,
+                        driver: ""
+                    }
+                ]
+            };
+            
+            let isResultImportedOrFromScratch = -1;
+
+            viewJSONData(json, season, points, tracks, constructor, driver, status, isResultImportedOrFromScratch);
+
+            disableNumberInputsAndAddMoreBtns();
+            enableEditOnErrorAfterLoading(json);
+        });
+
+        $('.homeBtn').click(function(event) {
+            location.reload(true);
+
+            $('.homeBtn').html('<i class="fas fa-sync fa-spin"></i>');
+        });
+
+        $('#backToEditScreen').click(function(event) {
+            $('#serverResponseScreen').addClass('hidden');
+            $('#editScreen').removeClass('hidden');
+
+            $('#submitJSON').html('Submit');
+        });
+    });
+
+    function parseJSONWithErrorHandling(readerResult) {
+        $('#JSONFormatErrorsAlert').slideUp(500);
+
+        try {
+            return JSON.parse(readerResult);
+        }
+        catch(err) {
+            $('#JSONFormatErrorMessage').html(`[${err.message}]`);
+            $('#JSLintHrefLink').attr('href', `https://jsonlint.com/?json=${readerResult}`);
+            $('#JSONFormatErrorsAlert').slideDown(500);
+        }
+    }
+
+    function checkJsonKeys(json) {
+        $('#missingResultsAlert').slideUp(500);
+        $('#missingTrackAlert').slideUp(500);
+        $('#positionKeyErrorAlert').slideUp(500);
+
+        if(json !== undefined) {
+            if(json.hasOwnProperty('track') && json.hasOwnProperty('results')) {
+                return 1;
+            } else if(json.hasOwnProperty('track') == false) {
+                $('#missingTrackAlert').slideDown(500);
+            } else {
+                $('#missingResultsAlert').slideDown(500);
+            }
+        }
     }
 
-    function drivernextFunction(){
-        index = index + 1;
-        document.getElementById("positionid").value = jsondata.results[index].position;
-        document.getElementById("constructorid").value = jsondata.results[index].constructor_id;
-        document.getElementById("drivernameid").value = jsondata.results[index].driver_id;
-        document.getElementById("fastesttimeid").value = jsondata.results[index].fastestlaptime;
-        document.getElementById("gridid").value = jsondata.results[index].grid;
-        document.getElementById("stopsid").value = jsondata.results[index].stops;
-        document.getElementById("timeid").value = jsondata.results[index].time;
+    function isAllPositionKeysPresentAndValidWithoutDuplicates(resultsArray) {
```

Comment:
_damn that is a long nameeee....._  😱

Reply:
Yes. Noob naming scheme. Pls guide me on how I can break it down.